lib/heroku-deflater/cache_control_manager.rb

```diff
@@ -20,9 +20,9 @@ def setup_max_age(max_age)
 
     def cache_control_headers
       if rails_version_5?
-        { 'Cache-Control' => cache_control }
+        { 'Cache-Control' => app.config.public_file_server.headers }
```

Comment:
Yeah, that's the type of thing I'm talking about. That would make your solution the better one. Other, non-cache-related headers are possible too, in theory, which should be preserved as well.

I think this should be merged with your changes included.